packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        // Store the configuration for this safe
+        safeConfigs[msg.sender] = _config;
+
+        // Clear any existing challenge when configuring/re-configuring
+        // If a challenge exists, it MUST be canceled, including emitting the appropriate events
+        if (challengeStartTime[msg.sender] != 0) {
+            delete challengeStartTime[msg.sender];
+            emit ChallengeCancelled(msg.sender);
+        }
+
+        emit ModuleEnabled(msg.sender, _config.livenessResponsePeriod, _config.fallbackOwner);
+    }
+
+    /// @notice Clears the module configuration for a Safe
+    /// @dev MUST only be executable by a Safe that has DISABLED this module first
+    /// @dev MUST erase the existing liveness_response_period and fallback_owner data related to the calling safe
+    /// @dev Note: Clearing the configuration also cancels any ongoing challenges
+    function clear() external {
+        // Check if the calling safe has configuration set
+        ModuleConfig storage config = safeConfigs[msg.sender];
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        // Check that this module is NOT enabled on the calling Safe
+        // This prevents clearing configuration while module is still enabled
+        Safe safe = Safe(payable(msg.sender));
+        if (safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleStillEnabled();
+        }
+
+        // Erase the configuration data for this safe
+        delete safeConfigs[msg.sender];
+        // Also clear any active challenge
+        delete challengeStartTime[msg.sender];
+        emit ModuleDisabled(msg.sender);
+    }
+
+    /// @notice Returns challenge_start_time + liveness_response_period if there is a challenge for the given safe, or
+    /// 0 if not
+    /// @dev MUST never revert
+    /// @param _safe The Safe address to query
+    /// @return The challenge end timestamp, or 0 if no challenge
+    function getChallengePeriodEnd(address _safe) external view returns (uint256) {
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            return 0;
+        }
+        ModuleConfig storage config = safeConfigs[_safe];
+        return startTime + config.livenessResponsePeriod;
+    }
+
+    /// @notice Challenges an enabled safe
+    /// @dev MUST only be executable by fallback owner of the challenged safe
+    /// @dev MUST revert if there is a challenge for the safe
+    /// @dev MUST set challenge_start_time to the current block time
+    /// @dev MUST emit the ChallengeStarted event
+    /// @param _safe The Safe to challenge
+    function challenge(address _safe) external {
+        ModuleConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
```

Comment:
We use this as a signal that the module isn't configured, probably nicer if we do an internal function _assertModuleConfigured() instead.